### ISSUE TITLE
Wiring

### ISSUE THREAD

**Issue by Vicentetomas** [gomez9656/ESP32CAM-ESP8266-NRF24L01Z-acknowledge#1]
Congrats for the code. Could you please specify the wiring in the ESP32cam and Esp8266? something I misunderstand and the nrf2401 fails
Thank you very much

**Comment by juanpgg-git** [comment]
Hi Vincent, thanks a lot for your feedback!

At the begining of each sketch you can find the pins:

//ESP32-CAM
//create a SPI with software
RHSoftwareSPI spi; 

// Singleton instance of the radio driver
RH_NRF24 driver(15, 2, spi); //CSN, CE, SPI

//ESP8266
// Singleton instance of the radio driver
RH_NRF24 driver(2, 4);  //CSN, CE

The RHSoftwareSPI library setups the other pins, and you are right, I didn't specificy those, but you can check it on the library :)

**Comment by SiliconEmpire** [comment]
Hi Gomez, i sincerely appreciate you for putting your time into this, av been stuck on this project for weeks coming across this repo is a life saver, i hope it work for me

**Comment by juanpgg-git** [comment]
Hi SiliconEmpire, 

I'm really happy this project can help solve yours :)

**Comment by SiliconEmpire** [comment]
Hi Gomez, i tried to run your code and its not working for me. from the camera side am getting this error from the serial monitor:

`change channel succeed
Camera init succeed
too big
too big
buffer length: 42602
Start
Start failed
 len failed
`

while from the esp8266 side am getting this error from the serial monitor:

`....
init failed
change channel succeed
`
please i need your help to fix this thanks.

**Comment by SiliconEmpire** [comment]
the problem is coming from the NRF24 module because am always getting 'nrf24 init failed` from both end

**Comment by juanpgg-git** [comment]
Hi SilliconEmpire,

from what I can see in "too big", the image is more significant than the buffer declared. You're using the same modules as in the example, or have you added more code on top of it? the camera in the esp32-cam is the same as in the example?

You're right about the NRF24 part. I suggest, double-checking the connections, and powering the nrf24 boards with a battery or a different power source than the esp8266 and esp32

**Comment by SiliconEmpire** [comment]
i have triple check my connections still getting, nrf24 init failed

**Comment by juanpgg-git** [comment]
can you try initializing the nrf24 in a simple sketch? just using the Radiohead library

**Comment by SiliconEmpire** [comment]
Alright ill do that, coding on vs code(platform io) i hope thats not a problem

**Comment by juanpgg-git** [comment]
It shouldn't, but just to make sure try it on Arduino, doesn't take much time

**Comment by SiliconEmpire** [comment]
hello bro i have done everything i could(power from external supply, added decoupling capacitor) still failing to initialise the NRF24, but initialising it on an Arduino work pretty well... please help

**Comment by juanpgg-git** [comment]
Hello SilliconEmpire, I wouldn't know where else to look at. I remember I had problems with the ESP32-CAM and SPI, but that was solved using the software SPI, and not the hardware SPI. 
On the ESP8266 side, I remember it worked with the library without a problem.
Maybe they updated it in the last year?

**Comment by SiliconEmpire** [comment]
Hello Gomez9656, thanks i really appreciate your time, i finally was able to get it initialised, but the program just get stuck at this point for the camera part
` nrf24 init succeed
change channel succeed
Camera init succeed
buffer length: 41779
Start`

while for the esp8266 its stuck here 
`
init succeed
change channel succeed
Start
41199
`
and i also observed that when i change to NRF24(the version with antenna) it just stop here 
`
nrf24 init succeed
change channel succeed
Camera init succeed
buffer length: 41538
Start
Start failed
 len failed

`
i would have loved to use the the one with antenna for better range, please any advice or hint for me?

**Comment by SiliconEmpire** [comment]
"Note: both your transmitter and receiver microcontrollers must be working at the same frequency, if not, the communication will last 10+ minutes." and what do you mean by the above statement, i hope am not a victim of it already, i thought the NRF is already on same frequency by default? am sorry for too many question, just wonna get this thing to work

**Comment by SiliconEmpire** [comment]
Please Help Bro

**Comment by gomez9656** [comment]
Hi

**Comment by juanpgg-git** [comment]
Hi there,

I was actually using the NRF24 with the antenna, so no idea why it is not working on your side.

About the frequency, I did this in the Arduino IDE. Just go to tools -> CPU frequency.
I suppose there's something similar on platform.io?

**Comment by SiliconEmpire** [comment]
Yes there should be something similar thanks I'll try that again

**Comment by SiliconEmpire** [comment]
Thanks Bro, I finally was able to get it to work, but please how do I get to view the picture at the url endpoint you provided....
 also I wanted to modify it such that I can post to telegram from the ESP8266 but the image buffer becomes empty after connecting to wifi, I don't know if the problem is with the telegram library, please any hint for this will be extremely helpful

**Comment by juanpgg-git** [comment]
You're right, I didn't post how to view the picture at the URL.

Please refer to: https://randomnerdtutorials.com/esp32-cam-http-post-php-arduino/

I think you will find something on how to do the telegram feature you mention
